### PR TITLE
Fix rebuild for python3.12 update

### PR DESCRIPTION
**Pull request by raulcd**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

I am just trying to fix the build from the auto-tick for Python 3.12

### CODE REVIEW

**Comment by raulcd** [comment]
ok, the error is solved but there are other issues related to the Python 3.12 builds. I currently don't have the time to investigate so feel free to close this PR and use or disregard my suggestion.

**Comment by jakirkham** [comment]
No need to close. This can be a useful point for someone else to get started 😉

Thanks for getting this work going! 🙏

File: recipe/meta.yaml [hunk 1]
```diff
@@ -27,6 +27,7 @@ requirements:
   host:
     - python
     - ncurses
+    - setuptools
```
Comment (isuruf) [inline recipe/meta.yaml:30]: You need this in run too. Also please restrict this for `py>=312`.

**Comment by jorisvandenbossche** [comment]
The python test on Python 3.12 gives the follow gdb output:

```
Program received signal SIGSEGV, Segmentation fault.
0x00007f807219c657 in kill () from /lib64/libc.so.6
Traceback (most recent call first):
  <optimized out>
  File "/tmp/tmpctni8yos/info/recipe/testing/process_to_debug.py", line 3, in <module>
    os.kill(os.getpid(), signal.SIGSEGV)
```

while comparing this with the Python 3.11 logs, I see:

```
Program received signal SIGSEGV, Segmentation fault.
0x00007fc0ff418657 in kill () from /lib64/libc.so.6
Traceback (most recent call first):
  <built-in method kill of module object at remote 0x7fc0ff191760>
  File "/tmp/tmpqvo6o9q7/info/recipe/testing/process_to_debug.py", line 3, in <module>
    os.kill(os.getpid(), signal.SIGSEGV)
  <built-in method kill of module object at remote 0x7fc0ff191760>
  File "/tmp/tmpqvo6o9q7/info/recipe/testing/process_to_debug.py", line 3, in <module>
  File "/tmp/tmpqvo6o9q7/info/recipe/testing/process_to_debug.py", line 3, in <module>
    os.kill(os.getpid(), signal.SIGSEGV)
Program received signal SIGSEGV, Segmentation fault.
```

I don't know if that is indicating a problem (if that output shouldn't have changed), but in case that change is fine (eg because of a Python change), the script asserting some things being present in the output probably needs to be updated. For example:

https://github.com/conda-forge/gdb-feedstock/blob/2707c0bcecee73a8009d1ac0b7a2640f39f78af2/recipe/run_test.sh#L50-L53

`built-in method kill` is no longer in the output.

Maybe the Python3.12 build on conda-forge has less debug symbols?

**Comment by Gunni** [comment]
FYI

Python 3.12.0 alpha 7

Release date: 2023-04-04

[gh-102809](https://github.com/python/cpython/issues/102809): Misc/gdbinit was removed.

**Comment by phil-blain** [comment]
> The python test on Python 3.12 gives the follow gdb output:
> 
> ```
> Program received signal SIGSEGV, Segmentation fault.
> 0x00007f807219c657 in kill () from /lib64/libc.so.6
> Traceback (most recent call first):
>   <optimized out>
>   File "/tmp/tmpctni8yos/info/recipe/testing/process_to_debug.py", line 3, in <module>
>     os.kill(os.getpid(), signal.SIGSEGV)
> ```
> 
> while comparing this with the Python 3.11 logs, I see:
> 
> ```
> Program received signal SIGSEGV, Segmentation fault.
> 0x00007fc0ff418657 in kill () from /lib64/libc.so.6
> Traceback (most recent call first):
>   <built-in method kill of module object at remote 0x7fc0ff191760>
>   File "/tmp/tmpqvo6o9q7/info/recipe/testing/process_to_debug.py", line 3, in <module>
>     os.kill(os.getpid(), signal.SIGSEGV)
>   <built-in method kill of module object at remote 0x7fc0ff191760>
>   File "/tmp/tmpqvo6o9q7/info/recipe/testing/process_to_debug.py", line 3, in <module>
>   File "/tmp/tmpqvo6o9q7/info/recipe/testing/process_to_debug.py", line 3, in <module>
>     os.kill(os.getpid(), signal.SIGSEGV)
> Program received signal SIGSEGV, Segmentation fault.
> ```
> 
> I don't know if that is indicating a problem (if that output shouldn't have changed), but in case that change is fine (eg because of a Python change), the script asserting some things being present in the output probably needs to be updated. For example:
> 
> https://github.com/conda-forge/gdb-feedstock/blob/2707c0bcecee73a8009d1ac0b7a2640f39f78af2/recipe/run_test.sh#L50-L53
> 
> `built-in method kill` is no longer in the output.
> 
> Maybe the Python3.12 build on conda-forge has less debug symbols?

Indeed, it seems the 3.12 build has less debug symbols. Simply moving the grep call for "builtin method kill" inside the `else` clause should be sufficient, along with listing 3.12 as undebugable:

```diff
diff --git a/recipe/run_test.sh b/recipe/run_test.sh
index da615bb..6c40e98 100644
--- a/recipe/run_test.sh
+++ b/recipe/run_test.sh
@@ -48,9 +48,6 @@ if [[ $(uname -m) == "ppc64le" ]]; then
 fi
 
 gdb -batch -ex "run" -ex "py-bt" --args python "$RECIPE_DIR/testing/process_to_debug.py" | tee gdb_output
-if [[ "$CONDA_PY" != "27" ]]; then
-    grep "built-in method kill" gdb_output
-fi
 
 # Unfortunately some python packages do not have enough debug info for py-bt
 #
@@ -64,7 +61,7 @@ fi
 # debugged out-of-the-box with this gdb package. When things change, there is not much to be
 # done besides adding or removing versions from this list.
 # Example: insufficient_debug_info_versions=("27" "37")
-insufficient_debug_info_versions=()
+insufficient_debug_info_versions=("312")
 
 if [[ " ${insufficient_debug_info_versions[@]} " =~ " ${CONDA_PY} " ]]; then
     if grep "line 3" gdb_output; then
@@ -75,6 +72,9 @@ if [[ " ${insufficient_debug_info_versions[@]} " =~ " ${CONDA_PY} " ]]; then
 else
     # We are lucky! This Python version has enough debug info for us to easily identify
     # the exact Python code where the crash happened.
+    if [[ "$CONDA_PY" != "27" ]]; then
+        grep "built-in method kill" gdb_output
+    fi
     grep "line 3" gdb_output
     grep "process_to_debug.py" gdb_output
     grep 'os.kill(os.getpid(), signal.SIGSEGV)' gdb_output
```

**Comment by phil-blain** [comment]
> What is the reason new gdb builds aren't produced in conda-forge? Is this the python 3.12 compatibility?

No, simply that I do not have a lot of time to maintain the feedstock, and the other maintainers are not really implicated anymore. Plus, I do not myself use the conda-forge build of GDB anymore as I find it's simpler to build it from scratch.

**Comment by mikev-junimo** [comment]
> > What is the reason new gdb builds aren't produced in conda-forge? Is this the python 3.12 compatibility?
> 
> No, simply that I do not have a lot of time to maintain the feedstock, and the other maintainers are not really implicated anymore. Plus, I do not myself use the conda-forge build of GDB anymore as I find it's simpler to build it from scratch.

Is there anything I can help with to push this forward?
We basically rely on conda-forge packages instead of distro packages so getting this working again would be great.
The problem is that I have zero knowledge of conda-forge packaging processes / best practices / etc...

**Comment by mikev-junimo** [comment]
Ah, looks like some work is being done in https://github.com/conda-forge/gdb-feedstock/pull/69

**Comment by phil-blain** [comment]
Thank you @raulcd, Python 3.12 was added in #70, I'll close this one.

**Pull request closed without merging**